airflow/contrib/operators/singularity_operator.py

```diff
@@ -0,0 +1,176 @@
+
+        # Pull the container if asked, and ensure not a binary file
+        if self.force_pull and not os.path.exists(self.image):
+            self.log.info('Pulling container %s', self.image)
+            image, lines = self.cli.pull(self.image, stream=True)
+            for line in lines:
+                self.log.info(line)
+
+            # Move the container to where it's desired
+            if self.pull_folder is not None:
+                self.image = os.path.join(self.pull_folder, os.path.basename(image))
+                shutil.move(image, self.image)
+
+        # Prepare list of binds
+        for bind in self.volumes:
+            self.options = self.options + ['--bind', bind]
+
+        # Does the user want a custom working directory?
+        if self.working_dir is not None:
+            self.options = self.options + ['--workdir', self.working_dir]
+
+        # Export environment before instance is run
+        for enkey, envar in self.environment.items():
+            self.log.debug('Exporting %s=%s', envar, enkey)
+            os.putenv(enkey, envar)
+            os.environ[enkey] = envar
+
+        # Create a container instance
+        self.log.debug('Options include: %s', self.options)
+        self.instance = self.cli.instance(self.image,
+                                          options=self.options,
+                                          args=self.start_command,
+                                          start=False)
+
+        self.instance.start()
+        self.log.info(self.instance.cmd)
+        self.log.info('Created instance %s from %s', self.instance, self.image)
+
+        self.log.info('Running command %s', self.get_command())
+        self.cli.quiet = True
+        result = self.cli.execute(self.instance,
+                                  self.get_command(),
+                                  return_result=True)
+
+        # Stop the instance
+        self.log.info('Stopping instance %s', self.instance)
+        self.instance.stop()
+
+        if self.auto_remove is True:
+            if os.path.exists(self.image):
+                shutil.rmtree(self.image)
+
+        # If the container failed, raise the exception
+        if result['return_code'] != 0:
+            message = result['message']
+            raise AirflowException('Singularity failed: %s' % message)
+
+        self.log.info('Output from command %s', result['message'])
+
+    def get_command(self):
+        if self.command is not None and self.command.strip().find('[') == 0:
+            commands = ast.literal_eval(self.command)
+        else:
+            commands = self.command
+        return commands
+
+    def on_kill(self):
+        if self.instance is not None:
+            self.log.info('Stopping Singularity instance')
+            self.instance.stop()
+
+            # If an image exists, clean it up
+            if self.auto_remove is True:
+                if os.path.exists(self.image):
```

Comment:
```suggestion
                if self.auto_remove and os.path.exists(self.image):
```

Reply:
Fixed this as well, thank you for the suggestion!